src/openms/source/FORMAT/HANDLERS/MzMLSpectrumDecoder.cpp

```diff
@@ -110,19 +110,32 @@ namespace OpenMS
 
     // TODO: handle meta data from the binaryDataArray tag -> currently ignored
     // since we have no place to store them
-
     OpenMS::Interfaces::BinaryDataArrayPtr intensity_array(new OpenMS::Interfaces::BinaryDataArray);
     OpenMS::Interfaces::BinaryDataArrayPtr x_array(new OpenMS::Interfaces::BinaryDataArray);
-    for (Size n = 0; n < default_array_length_; n++)
+    x_array->data.reserve(default_array_length_);
+    intensity_array->data.reserve(default_array_length_);
+    if (x_precision_64 && !int_precision_64)
```

Comment:
why do you couple coord and intensity?!
They are independent.
```
if (x_precision_64) x_array->data.insert(x_array->data.begin(), data_[x_index].floats_64.begin(), data_[x_index].floats_64.end());
else x_array->data.insert(x_array->data.begin(), data_[x_index].floats_32.begin(), data_[x_index].floats_32.end());
```

same for intensity. Done.